services/web/server/src/simcore_service_webserver/studies_dispatcher/_studies_access.py

```diff
@@ -8,70 +8,113 @@
     any template pre-assigned to a user
 
     Enforces a constraint: a user CANNOT have multiple copies of the same template
     """
-    new_uuid = str(uuid5(BASE_UUID, str(template_uuid) + str(user_id) + str(query)))
+    new_uuid = str(uuid5(_BASE_UUID, str(template_uuid) + str(user_id) + str(query)))
     return new_uuid
 
 
-async def get_public_project(app: web.Application, project_uuid: str):
+async def _get_published_template_project(
+    app: web.Application, project_uuid: str
+) -> ProjectDict:
     """
-    Returns project if project_uuid is a template and is marked as published, otherwise None
+    raises RedirectToFrontEndPageError
     """
     db = ProjectDBAPI.get_from_app_context(app)
-    prj, _ = await db.get_project(
-        -1, project_uuid, only_published=True, only_templates=True
-    )
-    return prj
 
+    try:
+        prj, _ = await db.get_project(
+            project_uuid=project_uuid,
+            # NOTE: these are the conditions for a published study
+            # 1. MUST be a template
+            only_templates=True,
+            # 2. MUST be checked for publication
+            only_published=True,
+            # 3. MUST be shared with EVERYONE=1 in read mode, i.e.
+            user_id=ANY_USER,  # any user
+            check_permissions="read",  # any user has read access
+        )
+        if not prj:
+            # Not sure this happens but this condition was checked before so better be safe
+            raise ProjectNotFoundError(project_uuid)
 
-async def create_temporary_user(request: web.Request):
-    """
-    TODO: user should have an expiration date and limited persmissions!
-    """
+        return prj
+
+    except (ProjectNotFoundError, ProjectInvalidRightsError) as err:
+        log.debug(
+            "Requested project with %s is not published. Reason: %s",
+            f"{project_uuid=}",
+            err.detailed_message(),
+        )
+
+        raise RedirectToFrontEndPageError(
+            MSG_PROJECT_NOT_PUBLISHED.format(project_id=project_uuid),
+            error_code="PROJECT_NOT_PUBLISHED",
+            status_code=web.HTTPNotFound.status_code,
+        ) from err
+
+
+async def _create_temporary_user(request: web.Request):
     from ..login.storage import AsyncpgStorage, get_plugin_storage
     from ..login.utils import ACTIVE, GUEST, get_client_ip, get_random_string
     from ..security_api import encrypt_password
 
     db: AsyncpgStorage = get_plugin_storage(request.app)
     redis_locks_client: aioredis.Redis = get_redis_lock_manager_client(request.app)
+    settings: StudiesDispatcherSettings = get_plugin_settings(app=request.app)
 
-    # TODO: avatar is an icon of the hero!
+    # Profile for temporary user
     random_uname = get_random_string(min_len=5)
     email = random_uname + "@guest-at-osparc.io"
     password = get_random_string(min_len=12)
+    expires_at = datetime.utcnow() + settings.STUDIES_GUEST_ACCOUNT_LIFETIME
```

Comment:
sure about using utcnow?

Reply:
for the moment, it only works with naive datetimes when using asyngpg. I still need to figure out why